server/middlewares/incident.js

```diff
@@ -1,337 +1,316 @@
-                status: 400,
-                message: 'Invalid URL'
-            });
-        }
-        const foundincident = db.incident.find(incident => incident.id === (incidentId));
-        if (!foundincident) {
-            return res.status(404).json({
-                status: 404,
-                message: 'This incident does not exist'
-            });
-        }
-        req.body.foundincident = foundincident;
-        return next();
+  /**
+   * Get a specific product
+   * @param {object} req - The request object
+   * @param {object} res - The response object
+   * @param {function} next - Calls the next function
+   * @returns {object} JSON representing the failure message
+   * @memberof IncidentValidator
+   */
+  static getOneIncident(req, res, next) {
+    const incidentId = parseInt(req.params.id, 10);
+    if (!incidentId) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Invalid ID'
+      });
+    }
+    const foundincident = db.incident.find(incident => incident.id === (incidentId));
+    if (!foundincident) {
+      return res.status(404).json({
+        status: 404,
+        message: 'This incident does not exist'
+      });
     }
+    req.body.foundincident = foundincident;
+    return next();
+  }
 
+  /**
+   * Check incident input
+   * @param {object} req - The request object
+   * @param {object} res - The response object
+   * @param {function} next - Calls the next function
+   * @returns {object} JSON representing the failure message
+   * @memberof IncidentValidator
+   */
+  static incidentInputChecker(req, res, next) {
+    const {
+      createdOn, createdBy, type, location, status, imageUrl, videoUrl
+    } = req.body;
+    const verifyCreatedBy = parseInt(createdBy, 10);
+    if (createdOn === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdOn cannot be undefined'
+      });
+    }
+    const createdOnVerifier = /^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/;
 
+    if (!createdOnVerifier.test(createdOn)) {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdOn format should be yyyy-mm-dd'
+      });
+    }
+    if (createdOn === '') {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdOn should have a date'
+      });
+    }
 
-    /**
-       * Check incident input
-       * @param {object} req - The request object
-       * @param {object} res - The response object
-       * @param {function} next - Calls the next function
-       * @returns {object} JSON representing the failure message
-       * @memberof IncidentValidator
-     */
-    static incidentInputChecker(req, res, next) {
-        let {
-            createdOn, createdBy, type, location, status, imageUrl, videoUrl
-        } = req.body;
+    if (createdBy === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdBy cannot  be undefined'
```

Comment:
You should not have the word `undefined` in your error messages. Use something like `createdBy is required`

Reply:
Implemented and resolved